### ISSUE TITLE
Indentation problem with converting to fonts definition like pubspec.yaml

### ISSUE THREAD

**Issue by SuperPenguin** [alexei-sintotski/json2yaml#17]
I'm trying to write [fonts definition similar to pubspec.yaml](https://flutter.dev/docs/cookbook/design/fonts#2-declare-the-font-in-the-pubspec), and somehow the result got the indentation wrong.
Luckily a formatter able to fix the indentation for me, so it's not a big bug for me.

code

```dart
import 'dart:io';
import 'package:json2yaml/json2yaml.dart';

void main(List<String> args) async {
  Map<String, dynamic> data = {
    "flutter": {
      "fonts": [
        {
          "family": "Poppins",
          "fonts": [
            {
              "asset": "fonts/Poppins/Poppins-Thin.ttf",
              "weight": 100,
            },
            {
              "asset": "fonts/Poppins/Poppins-ThinItalic.ttf",
              "weight": 100,
              "style": "italic",
            },
          ],
        },
      ],
    },
  };

  if (File('testyaml.yaml').existsSync()) {
    File('testyaml.yaml').deleteSync();
  }

  File('testyaml.yaml').createSync();
  File('testyaml.yaml').writeAsStringSync(
    json2yaml(
      data,
      yamlStyle: YamlStyle.pubspecYaml,
    ),
  );
}
```

result

```yaml
flutter:
  fonts:
    -
          family: Poppins
          fonts:
            -
                  asset: "fonts/Poppins/Poppins-Thin.ttf"
                  weight: 100
            -
                  asset: "fonts/Poppins/Poppins-ThinItalic.ttf"
                  weight: 100
                  style: italic
```

expected

```yaml
flutter:
  fonts:
    - family: Poppins
      fonts:
        - asset: "fonts/Poppins/Poppins-Thin.ttf"
          weight: 100
        - asset: "fonts/Poppins/Poppins-ThinItalic.ttf"
          weight: 100
          style: italic
```

Also thank you for this amazing package. Huge time saver :+1: for editing yaml

**Comment by alexei-sintotski** [comment]
@SuperPenguin I just released version 3.0.0 supporting correct rendering of nested structures. Could you please check whether this issue is fixed?

**Comment by SuperPenguin** [comment]
> @SuperPenguin I just released version 3.0.0 supporting correct rendering of nested structures. Could you please check whether this issue is fixed?

Yep the structure and indentation now match pubspec style
